tests/test_base_script.py

```diff
@@ -82,6 +91,10 @@ def setUp(self):
 
         Raises
         ------
+        AssertionError
+            If the script's ``config``, log level or checkpoints
```

Comment:
I am quite sure I want this `Raises` section because it makes it obvious that this method actually tests that configuration worked as requested rather than just trying and not testing.

Reply:
I would have made that clear in the primary docstring: "Configure and test a script by calling ``do_configure``and checking that the scripts subsequent configuration is correct". I'd also name the method either `_test_*` or `check_*`, so that it's clear from the name.

Reply:
OK. And I'll rename it to `configure_and_check`

Reply:
We resolved this in conversation.